infra/blueprint-test/pkg/tft/terraform.go

```diff
@@ -481,22 +488,50 @@ func (b *TFBlueprintTest) DefaultApply(assert *assert.Assertions) {
 
 // Init runs the default or custom init function for the blueprint.
 func (b *TFBlueprintTest) Init(assert *assert.Assertions) {
+	// allow only single write as Terraform plugin cache isn't concurrent safe
+	if err := b.tftCacheMutex.Lock(); err != nil {
+		b.t.Fatalf("Could not acquire lock: %v", err)
+	}
 	b.init(assert)
+	if err := b.tftCacheMutex.Unlock(); err != nil {
+		b.t.Fatalf("Could not release lock: %v", err)
+	}
 }
 
 // Apply runs the default or custom apply function for the blueprint.
 func (b *TFBlueprintTest) Apply(assert *assert.Assertions) {
+	// allow only parallel reads as Terraform plugin cache isn't concurrent safe
```

Comment:
Yes, but other than for `terraform init` we are using read locks as unlimited read locks are permitted parallel.  Pretty much any terraform call can fail if a `terraform init` process is actively modifying the provider cache at the wrong moment.  So this will prevent any `terraform init` from be run at the same as other Terraform calls.

Reply:
Good point, that makes sense!